remotewrite/sender/README.md

```diff
@@ -1,40 +1,112 @@
+- Float samples encoding
+- Native Histograms encoding
+- Exemplars encoding
+- Protocol headers and content negotiation
+- Error handling and retry logic
+- Backoff and batching behavior
+- Metadata and symbol table management
+- Request formatting and compression
 
-```sh
-$ go test --tags=compliance -v ./
+## Limitations
+
+The test suite validates the format and structure of requests sent by the sender but does not verify end-to-end data flow or persistence. Tests examine:
+
+- Request payload structure and encoding
+- Protocol compliance of generated requests
+- Proper header usage and content negotiation
+- Correct retry and backoff behavior when receivers respond with errors
+
+Because senders may have different configuration options and capabilities, passing all tests does not guarantee a sender supports every Remote-Write feature (such as Native Histograms). Some senders may not expose certain features or may require specific configuration.
+
+You should review the detailed test output to judge compliance for your sender. A successful `go test` run demonstrates the sender correctly encodes and sends Remote-Write v2 requests when configured to do so.
+
+## Prerequisites
+
+- Go 1.23 or later
+- The sender binary to test (e.g., Prometheus)
+
+The test suite automatically downloads and runs Prometheus as the reference sender implementation. For testing custom senders, place the binary in the `bin/` directory.
+
+## Configuration
+
+The test suite uses environment variables for configuration:
+
+### Sender Selection
+
+You can specify which sender to test using the `PROMETHEUS_RW2_COMPLIANCE_SENDERS` environment variable:
+
+```bash
+export PROMETHEUS_RW2_COMPLIANCE_SENDERS="prometheus"
+go test -v
 ```
 
-To run all the tests for a single target:
+Currently supported senders:
+- `prometheus` - The reference Prometheus implementation (automatically downloaded)
+
+### Test Timeout
 
-```sh
-$ go test --tags=compliance -run "TestRemoteWrite/prometheus/.+" -v ./
+You can override the default test timeout using:
+
+```bash
+export PROMETHEUS_RW2_COMPLIANCE_TEST_TIMEOUT="10m"
+go test -v
+```
+
+## Running Tests
+
+The test suite automatically sets up mock receiver endpoints and forks sender instances. You can run tests using standard Go test commands:
+
+```bash
+# Run all compliance tests
+go test -v -timeout 5m
+
+# Run tests for a specific area
+go test -v -run TestHistograms
+go test -v -run TestExemplars
+
+# Run specific test case
+go test -v -run "TestExemplarEncoding/exemplar_with_trace_id"
+
+# Run tests with detailed output
+go test -v -count=1
+
+# Run tests against specific sender only
+PROMETHEUS_RW2_COMPLIANCE_SENDERS="prometheus" go test -v
 ```
 
-To run a single test across all the targets:
+## HTML visualisation
 
-```sh
-$ go test --tags=compliance -run "TestRemoteWrite/.+/Counter" -v ./
+This repository contains a `index.html` that enables viewing the results of the compliance tests.
```

Comment:
Let's move it locally back or fix both receiver and sender docs. 

In fact this README is almost 1:1 with receiver. Probably easy to combine both docs???